test/controllers/file-patch-controller.test.js

```diff
@@ -96,6 +96,18 @@ describe('FilePatchController', function() {
 
       getFilePatchForPath = sinon.stub(repository, 'getFilePatchForPath');
     });
+    describe('buildURI', function() {
```

Comment:
@smashwilson do you think it would be worth it to also add tests at the react component layer?   IE, clicking on staged changes and verifying that the filepath is correct?  Or is this sufficient?

Reply:
This should be fine. I'd say that in this case, _good test coverage of the `buildURI()` method with wacky URIs + good test coverage of launching pane items = good test coverage of launching pane items with wacky URIs_. In other words, adding an integration-ish test for this particular loophole wouldn't really catch anything that that wouldn't.

:shipit:

Reply:
![bitmoji](https://render.bitstrips.com/v2/cpanel/55f5faaa-bbee-4e30-bb67-094748197f8f-f48e36cc-8fe7-4833-b594-88e139dc1265-v1.png?transparent=1&palette=1&width=246)

thank you!